File: examples/python3/handler.py
def handle(event, context):
    """handle a request to the function
    Args:
        event (dict): request params
        context (dict): function call metadata
    """

    return {
        "body": "Hello From Python3 runtime on Serverless Framework and Scaleway Functions",
        "headers": {
            "Content-Type": ["text/plain"],
        }
    }

# run 'pip install scaleway_functions_python' if necessary
if __name__ == "__main__":
    # The import is conditional so that you do not need
    # to package the library when deploying on Scaleway Functions.
    from scaleway_functions_python import local
    local.serve_handler(handle, port=8080)

File: examples/typescript/handler.ts
import {Callback, Context, Event} from "@scaleway/serverless-functions/framework/dist/types/types";

export {handle};

function handle(event: Event, context: Context, cb: Callback) {
    return {
        body: "Hello world!",
        headers: { "Content-Type": ["application/json"] },
        statusCode: 200,
    };
};

/* This is used to test locally and will not be executed on Scaleway Functions */
if (process.env.NODE_ENV === 'test') {
    import("@scaleway/serverless-functions").then(scw_fnc_node => {
        scw_fnc_node.serveHandler(handle, 8080);
    });
}
